content/doc/tutorials/building-a-python-app-with-pyinstaller.adoc

```diff
@@ -0,0 +1,486 @@
+least one of the values happens to be a string, the "add" function treats both
+values as a string and instead concatenates them together. The add function is
+accompanied by a set of unit tests to check that the function works as expected.
+
+*Duration:* This tutorial takes 20-40 minutes to complete (assuming you've
+already met the <<prerequisites,prerequisites>> below). The exact duration will
+depend on the speed of your machine and whether or not you've already
+<<run-jenkins-in-docker,run Jenkins in Docker>> from link:..[another
+introductory tutorial].
+
+You can stop this tutorial at any point in time and continue from where you left
+off.
+
+If you've already run though link:..[another tutorial], you can skip the
+<<prerequisites,Prerequisites>> and <<run-jenkins-in-docker,Run Jenkins in
+Docker>> sections below and proceed on to <<fork-sample-repository,forking the
+sample repository>>. (Just ensure you have
+link:https://git-scm.com/downloads[Git] installed locally.) If you need to
+restart Jenkins, simply follow the restart instructions in
+<<stopping-and-restarting-jenkins,Stopping and restarting Jenkins>> and then
+proceed on.
+
+include::doc/tutorials/_prerequisites.adoc[]
+** link:https://git-scm.com/downloads[Git] and optionally
+   link:https://desktop.github.com/[GitHub Desktop]
+
+include::doc/tutorials/_run-jenkins-in-docker.adoc[]
+
+
+[[fork-sample-repository]]
+=== Fork and clone the sample repository on GitHub
+
+Obtain the simple "add" Python application from GitHub, by forking the sample
+repository of the application's source code into your own GitHub account and
+then cloning this fork locally.
+
+. Ensure you are signed in to your GitHub account. If you don't yet have a
+  GitHub account, sign up for a free one on the https://github.com/[GitHub
+  website].
+. Fork the
+  https://github.com/jenkins-docs/simple-python-pyinstaller-app[`simple-python-pyinstaller-app`]
+  on GitHub into your local GitHub account. If you need help with this process,
+  refer to the https://help.github.com/articles/fork-a-repo/[Fork A Repo]
+  documentation on the GitHub website for more information.
+. Begin cloning your forked `simple-python-pyinstaller-app` repository (on
+  GitHub) locally to your machine by opening up a terminal/command line window
+  to the directory in which the repository will be cloned. This tutorial assumes
+  that `simple-python-pyinstaller-app` will be cloned within the following
+  directories on these platforms, which you should `cd` into now:
+* macOS - `/Users/<your-username>/Documents/GitHub`
+* Linux - `/home/<your-username>/GitHub`
+* Windows - `C:\Users\<your-username>\Documents\GitHub`
+
++
+where `<your-username>` is your user account's name on your operating system.
+
++
+*Note:*
+
+** On a Windows machine, use a Git bash command line window (as opposed to the
+   usual Microsoft command prompt).
+** Alternatively, and in particular if you have the GitHub Desktop app installed
```

Comment:
The phrasing of this sentence makes it sound almost as if the GitHub Desktop app is optional for clicking the green "Clone and Download" button.

I suggest dropping "and in particular"

The text about "following the directions" under the green button aren't necessary, as that's effectively what the following line tells the user to do. So I believe that  can be removed

Reply:
I decided to rework these instructions by splitting the GitHub Desktop and command line "cloning" instructions into separate/alternative bullet points.

Attempting to keep them combined made things confusing - and I was never completely happy with how these instructions originally were.